tests/conftest.py

```diff
@@ -173,24 +173,45 @@ def reset_tables(engine: Engine) -> None:
 
 
 @pytest.fixture(scope="session")
```

Comment:
The changes in this file are an unrelated issue that came up while working on this PR. The database engine and sessionmaker were created separately, and if you created the engine but not the sessionmaker, tests started to fail in surprising ways. This made the tests pass or fail depending on which order you ran them in. Took me about 3 hours to hunt this down... We now create both at the same time.